plugins/workspace-plugin/devfile.yaml

```diff
@@ -0,0 +1,72 @@
+apiVersion: 1.0.0
```

Comment:
I'm not sure it's a good practice to introduce devfiles in each directory ?

Reply:
This a devfile to work on the workspace plugin, as this project lives in a subdirectory of the git repo, it looks good to have it there, next to the package.json, etc ...

Reply:
but it's part of a yarn workspace so you should use top-level folder/definitions/devfile anyway to inherits from all workspace definition

Reply:
ok but isn't it overkill when a dev just want to work on 1 plugin ?

Reply:
It might be useful to have the devfile here, even just for reference purposes. It could be useful to point for new contributors in the future.

Reply:
On a maven project you won't have devfile all over the places in every modules as well
Also we do need parent package.json
It would be nightmare to maintain all these devfiles

Reply:
> It would be nightmare to maintain all these devfiles

That's a good point, better leave it out then @sunix

Reply:
no but the devfiles are maintained as their are used. And for maven submodules, it really depends if a submodule is to be used independently. Take the old che6 maven structure, we had a plugin folder. each plugin could be built independently.

Reply:
I'll have to add my voice to the detractors here: I don't think there is a benefit to having a devfile for every single che-theia plugin: you can already rebuild single plugins and I don't really see any other upside, but a maintenance burden: each time we change something in the recommended plugin development process, we'll have to go over all those devfiles. We're not doing a good job keeping the devfiles in sync with best practices.
As for extensions: you have to build all of those together anyway, as changes in one may break others (compile-time):

Reply:
it looks like: hey the current devfiles are not so good so let's add a new one. I mean, let's take a contributor. What devfile pick up if I find there are 30 devfiles across different folders ?

Depending on working on a che-theia task, I could work only on one extension and one plug-in, so it would result in a new submitted devfile to work with these two modules ?

devfile can use generic commands based on the current selected folder, so if you work on plug-ins basically they all have same commands: lint, compile, format, test, etc.

Reply:
This devfile is for coding the workspace plugin without having to rebuild che-theia and all the other plugins. It doesn't make sense to have it elsewhere.

Reply:
The maintenance issue is not related to the location of the devfiles. but how a devfile could be extended: so the devfile for the workspace plugin (that would be in the workspace plugin folder) could extend a more generic devfile for built-in plugins but adding its own specifics (for instance, custom commands, etc ...).
The devfile for the workspace plugin has to stay here because it would setup the right workspace for the workspace-plugin developer flow.

Reply:
and then if the suggestion is to move that devfile to `devfiles` I don't see how easier it would be to maintain. Or is the suggestion not to have devfile at all and let all the developers loosing time to setup an dev environment to work on each individual plugins?

Reply:
No, the suggestion is to have a a single devfile for developing che-theia.

Reply:
there is that devfile https://github.com/eclipse/che-theia/blob/master/devfiles/che-theia-all.devfile.yaml but it is doing way too many things when it is to work a a particular plugin.